assimp: Record imported actors from internals

exbluesbreaker opened the pull request:
Importer itself did't have actor stored which prevented metadata retrieval for display

Issues with missing metadata was reported in https://github.com/f3d-app/f3d/issues/1945

exbluesbreaker commented:
I am not exactly sure if on older version of vtk any fix is needed. I added guard for older vtk. f3d gets metadata in a different way in this case, but I do not have such vtk to test with, maybe some other fix is needed, but I cannot test it.

mwestphal commented:
> I am not exactly sure if on older version of vtk any fix is needed. I added guard for older vtk. f3d gets metadata in a different way in this case, but I do not have such vtk to test with, maybe some other fix is needed, but I cannot test it.

That looks correct! Can you add a test for it though ? A verbose + REGEXP test should do the trick

exbluesbreaker commented:
> > I am not exactly sure if on older version of vtk any fix is needed. I added guard for older vtk. f3d gets metadata in a different way in this case, but I do not have such vtk to test with, maybe some other fix is needed, but I cannot test it.
> 
> That looks correct! Can you add a test for it though ? A verbose + REGEXP test should do the trick

I'll do it tomorrow

exbluesbreaker commented:
So I added a simple test for number of actors, it is not super excisting as it is usually 1, but I didn't find a way to get number of cells or points

mwestphal commented on application/testing/CMakeLists.txt:
```diff
@@ -649,6 +649,10 @@ if(F3D_PLUGIN_BUILD_ASSIMP)
   endif()
 
+  if(VTK_VERSION VERSION_GREATER_EQUAL 9.3.20240707)
+    f3d_test(NAME TestAssimpMetaData DATA 16bit.fbx ARGS --verbose --load-plugins=assimp REGEXP "Number of actors: 1" NO_BASELINE)
```
See this test:

```
f3d_test(NAME TestVerbose DATA dragon.vtu ARGS --verbose REGEXP "Number of points: 13268\nNumber of cells: 26532" NO_BASELINE)
```

You cant do that same for the assimp data ?

exbluesbreaker replied:
> You cant do that same for the assimp data ?

Let me have a look, probably something else is then needed for supporting metadata output to console, output in GUI seems working. In console I only saw actors and files after current implementation of fix.

mwestphal replied:
indeed, you are right

exbluesbreaker commented:
I updated `Internals->Description` with points and cells data to see in on console output. I also updated test case for more points model so number is more "specific"

mwestphal commented:
Ha right I forgot that this API is a bit all over the place. In any case its a good fix.

exbluesbreaker commented on application/testing/CMakeLists.txt:
```diff
@@ -649,6 +649,10 @@ if(F3D_PLUGIN_BUILD_ASSIMP)
 
+  if(VTK_VERSION VERSION_GREATER_EQUAL 9.3.20240707)
+    f3d_test(NAME TestAssimpMetaData DATA duck.fbx ARGS --verbose --load-plugins=assimp REGEXP "Number of points: 12636" NO_BASELINE)
+  endif()
```
> Then you are not testing the metadata but the verbose mode. So I guess you dont need the VTK_VERSION check and the test should be named TestAssimpVerbose.
> 
> Lets not add a specific test for assimp metadata using the UI, that doesnt seems worth it.

Good point. I guess there should be some way to interact with GUI and retrieve text from it, but I didn't have much experience with advanced GUI programming.

exbluesbreaker replied:
> > I guess there should be some way to interact with GUI and retrieve text from it, but I didn't have much experience with advanced GUI programming.
> 
> Actually if you want to add a GUI test, its pretty simple, basically we just capture the image and compare it against a baseline :)
> 
> See
> 
> ```
> f3d_test(NAME TestMetaDataImporter DATA BoxAnimated.gltf ARGS -m UI)
> ```

Understood, will have a look

exbluesbreaker commented:
Also branch name is a bit misleading I noticed it only recently, it should be assimp not just fbx fix

mwestphal commented:
> Also branch name is a bit misleading I noticed it only recently, it should be assimp not just fbx fix

Dont worry about the branch name :)

exbluesbreaker commented:
Thanks for the tip about GUI test, I added it, now it seems complete

mwestphal commented:
Nice! lets wait fox @Meakk review before merging though.

mwestphal commented:
Thanks for the contribution @exbluesbreaker !
